### PR TITLE
[Arcane Mage] Add Spec

### PR DESCRIPTION
**Pull request by Sharrq**
Added initial basic support for Arcane.

### CODE REVIEW

File: src/Parser/Mage/Arcane/Modules/Features/Abilities.js [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+          suggestion: true,
+          recommendedEfficiency: 0.90,
+        },
+      },
+      {
+        spell: SPELLS.CHARGED_UP_TALENT,
+        category: Abilities.SPELL_CATEGORIES.ROTATIONAL_AOE,
+        cooldown: haste => 12 / (1 + haste),
+        enabled: combatant.hasTalent(SPELLS.LIVING_BOMB_TALENT.id),
+      },
```
Comment (CollCrom) [inline src/Parser/Mage/Arcane/Modules/Features/Abilities.js]: Should this be living bomb? You have charged up talent twice here

Comment (Sharrq) [reply src/Parser/Mage/Arcane/Modules/Features/Abilities.js]: woops, missed that. should be fixed now

File: src/Parser/Mage/Arcane/Modules/Features/AlwaysBeCasting.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    if (!boss || !boss.fight.disableDowntimeSuggestion) {
+      when(deadTimePercentage).isGreaterThan(0.2)
+        .addSuggestion((suggest, actual, recommended) => {
+          return suggest(<span>Your downtime can be improved. Try to Always Be Casting (ABC), try to reduce the delay between casting spells. If you have to move, try casting <SpellLink id={SPELLS.SCORCH.id} />.</span>)
+            .icon('spell_mage_altertime')
+            .actual(`${formatPercentage(actual)}% downtime`)
+            .recommended(`<${formatPercentage(recommended)}% is recommended`)
+            .regular(recommended + 0.15).major(recommended + 0.2);
+        });
```
Comment (CollCrom) [inline src/Parser/Mage/Arcane/Modules/Features/AlwaysBeCasting.js]: I would also suggest changing this to a getter that returns a suggestions thresholds object. Makes it easier to do an eventual checklist but that might just be my preference.

**Comment by CollCrom** [comment]
thanks =]

**Pull request merged**